client/rest-high-level/src/main/java/org/elasticsearch/client/indexlifecycle/DeleteLifecycleRequest.java

```diff
@@ -0,0 +1,55 @@
+/*
```

Comment:
Should this be Delete`Index`LifecycleRequest ?

Reply:
From what I saw, the classes with names including `IndexLifecycle` seem to relate to index<->lifecycle policy relationships (i.e. lifecycle `A` applies to index `B`), but ones with just `Lifecycle` relate to the management of lifecycles.  If that's correct, this class is named correctly.

@dakrone, can you confirm?

Reply:
I think this should be `DeleteLifecyclePolicyRequest` to indicate that the policy is going to be removed, and then `DeleteIndexLifecycleRequest` is for when we are removing a lifecycle from an index